examples/cholera_voronoi/agents.py

```diff
@@ -0,0 +1,83 @@
+from mesa.experimental.cell_space import CellAgent
+
+SUSCEPTIBLE = 0
+INFECTIOUS = 1
+REMOVED = 2
+
+
+class Person(CellAgent):
+    def __init__(self, unique_id, model, mortality_chance, recovery_chance):
+        super().__init__(unique_id, model)
+        self.state = SUSCEPTIBLE
+        self.mortality_chance = mortality_chance
+        self.recovery_chance = recovery_chance
+
+    def step(self):
+        if self.state == REMOVED:
+            return
+
+        if (
+            self.state == INFECTIOUS
+            and self.model.random.random() < self.recovery_chance
+        ):
+            self.state = SUSCEPTIBLE
+            self.model.infectious -= 1
+            self.model.susceptible += 1
+
+        if (
+            self.state == INFECTIOUS
+            and self.model.random.random() < self.mortality_chance
+        ):
+            self.state = REMOVED
+            self.model.infectious -= 1
+            self.model.removed += 1
+
+
+class Pump(CellAgent):
+    def __init__(
+        self,
+        unique_id,
+        model,
+        contaminated,
+        pumps_person_contamination_chance,
+        pumps_neighbor_contamination_chance,
+        cases_ratio_to_fix_pump,
+    ):
+        super().__init__(unique_id, model)
+        self.state = contaminated
+        self.pumps_person_contamination_chance = pumps_person_contamination_chance
+        self.pumps_neighbor_contamination_chance = pumps_neighbor_contamination_chance
+        self.cases_ratio_to_fix_pump = cases_ratio_to_fix_pump
+
+    def step(self):
+        if self.state is INFECTIOUS:
+            # Infect people in the cell
+            people = [
+                obj
+                for obj in self.cell.agents
+                if isinstance(obj, Person) and obj.state is not REMOVED
```

Comment:
Shouldn’t agents be just removed from cell.agents instead of having a state “removed”?